### PR TITLE
feat: add nix flake

### PR DESCRIPTION
**Pull request by Jappie3**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,6 +52,40 @@ hyprpm enable dynamic-cursors
...
+};
+```
+
+or add it like this:
+
+```nix
+wayland.windowManager.hyprland = {
+    enable = true;
+    extraConfig = ''
+        plugin = ${inputs.hypr-dynamic-cursors.packages.${pkgs.system}.hypr-dynamic-cursors}/lib/liblibhypr-dynamic-cursors.so
```
Comment (VirtCode) [inline README.md:84]: is `liblib...` intended here? looks like a typo

File: flake.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  outputs = {
+    self,
+    nixpkgs,
+    ...
+  } @ inputs: let
+    forAllSystems = function:
+      nixpkgs.lib.genAttrs [
+        "x86_64-linux"
+        "aarch64-linux"
```
Comment (VirtCode) [inline flake.nix:17]: this plugin makes heavy use of function hooks, which are not supported on arm iirc

**Review by VirtCode — COMMENTED** [review]
hope those comments make sense
otherwise LGTM as far as I can tell

File: README.md (hunk 1 above)
Comment (Jappie3) [reply README.md:84]: oh god ye I hastily pasted something there at the end, it should be `libhypr-dynamic-cursors.so`

File: flake.nix (hunk 2 above)
Comment (Jappie3) [reply flake.nix:17]: hm I haven't tested arm either, so I'll just remove that for now

**Comment by Jappie3** [comment]
the pleasure was mine (:

**Pull request merged**